src/utils/getTicks.ts

```diff
@@ -1,37 +1,26 @@
 import {scaleUtc} from 'd3-scale'
 import {ticks} from 'd3-array'
 
-import {ColumnType} from '../types'
-import {getTimeFormatter} from './getTimeFormatter'
-import {TextMetrics} from './getTextMetrics'
+import {Formatter} from '../types'
 
 const TICK_DENSITY = 0.5
 
 export const getTicks = (
   domain: number[],
-  length: number,
-  orientation: 'vertical' | 'horizontal',
-  columnType: ColumnType,
-  charMetrics: TextMetrics
+  rangeLength: number,
+  charLength: number,
+  formatter?: Formatter
 ): number[] => {
-  let sampleTick
-
-  if (columnType === 'time') {
-    sampleTick = getTimeFormatter(domain)(domain[1])
-  } else {
-    sampleTick = String(domain[1])
-  }
+  const sampleTick = formatter(domain[1])
+  const numTicks = getNumTicks(sampleTick, rangeLength, charLength)
 
-  const charLength =
-    orientation === 'vertical' ? charMetrics.height : charMetrics.width
+  switch (formatter._GIRAFFE_FORMATTER_TYPE) {
+    case 'TIME':
+      return getTimeTicks(domain, rangeLength, numTicks)
 
-  const numTicks = getNumTicks(sampleTick, length, charLength)
-
-  if (columnType === 'time') {
-    return getTimeTicks(domain, length, numTicks)
+    default:
```

Comment:
We can solve https://github.com/influxdata/giraffe/issues/79 here by adding another case statement.